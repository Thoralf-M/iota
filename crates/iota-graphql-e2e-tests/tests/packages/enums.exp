processed 8 tasks

init:
A: object(0,0)

task 1, lines 7-18:
//# publish --upgradeable --sender A
created: object(1,0), object(1,1)
mutated: object(0,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 5768400,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, line 20:
//# create-checkpoint
Checkpoint created: 1

task 3, lines 22-67:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x1fd21c18344f3bca811ca0cc1a9b4b4a0c2cbbc3aa2d08a100d4a53bb5a2a5ba",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0x29a4665c0e1dbec0ee7fc5f2881a0d01cd9c6793b43304817bdeef3fa04dfb0c",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0x3fde33fcf118358b1d9b357e9a7b5fbf8089851150e3014d8fffc42dde76faee",
                    "asMovePackage": {
                      "module": {
                        "enum": {
                          "name": "S",
                          "abilities": [
                            "COPY",
                            "DROP"
                          ],
                          "typeParameters": [],
                          "variants": [
                            {
                              "name": "V1",
                              "fields": [
                                {
                                  "name": "pos0",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            },
                            {
                              "name": "V2",
                              "fields": [
                                {
                                  "name": "x",
                                  "type": {
                                    "repr": "bool",
                                    "signature": {
                                      "ref": null,
                                      "body": "bool"
                                    }
                                  }
                                },
                                {
                                  "name": "y",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            }
                          ]
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 4, lines 69-82:
//# upgrade --package P0 --upgrade-capability 1,0 --sender A
created: object(4,0)
mutated: object(0,0), object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 6642400,  storage_rebate: 2606800, non_refundable_storage_fee: 0

task 5, line 84:
//# create-checkpoint
Checkpoint created: 2

task 6, lines 86-149:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x1fd21c18344f3bca811ca0cc1a9b4b4a0c2cbbc3aa2d08a100d4a53bb5a2a5ba",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0x29a4665c0e1dbec0ee7fc5f2881a0d01cd9c6793b43304817bdeef3fa04dfb0c",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0x2f142eac356a61786c3efdc6c2bb7387cbb892019e6449cb1ec1d8dc51228430",
                    "asMovePackage": {
                      "module": {
                        "s": {
                          "module": {
                            "package": {
                              "address": "0x3fde33fcf118358b1d9b357e9a7b5fbf8089851150e3014d8fffc42dde76faee"
                            }
                          },
                          "name": "S",
                          "abilities": [
                            "COPY",
                            "DROP"
                          ],
                          "typeParameters": [],
                          "variants": [
                            {
                              "name": "V1",
                              "fields": [
                                {
                                  "name": "pos0",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            },
                            {
                              "name": "V2",
                              "fields": [
                                {
                                  "name": "x",
                                  "type": {
                                    "repr": "bool",
                                    "signature": {
                                      "ref": null,
                                      "body": "bool"
                                    }
                                  }
                                },
                                {
                                  "name": "y",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            }
                          ]
                        },
                        "t": {
                          "module": {
                            "package": {
                              "address": "0x2f142eac356a61786c3efdc6c2bb7387cbb892019e6449cb1ec1d8dc51228430"
                            }
                          },
                          "name": "T",
                          "abilities": [],
                          "typeParameters": [
                            {
                              "constraints": [
                                "DROP"
                              ],
                              "isPhantom": false
                            }
                          ],
                          "variants": [
                            {
                              "name": "VV",
                              "fields": [
                                {
                                  "name": "y",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                },
                                {
                                  "name": "s",
                                  "type": {
                                    "repr": "0x3fde33fcf118358b1d9b357e9a7b5fbf8089851150e3014d8fffc42dde76faee::m::S",
                                    "signature": {
                                      "ref": null,
                                      "body": {
                                        "datatype": {
                                          "package": "0x3fde33fcf118358b1d9b357e9a7b5fbf8089851150e3014d8fffc42dde76faee",
                                          "module": "m",
                                          "type": "S",
                                          "typeParameters": []
                                        }
                                      }
                                    }
                                  }
                                },
                                {
                                  "name": "u",
                                  "type": {
                                    "repr": "$0",
                                    "signature": {
                                      "ref": null,
                                      "body": {
                                        "typeParameter": 0
                                      }
                                    }
                                  }
                                }
                              ]
                            }
                          ]
                        },
                        "v": {
                          "name": "V",
                          "variants": [
                            {
                              "name": "V",
                              "fields": [
                                {
                                  "name": "t",
                                  "type": {
                                    "repr": "0x3fde33fcf118358b1d9b357e9a7b5fbf8089851150e3014d8fffc42dde76faee::m::T<0x3fde33fcf118358b1d9b357e9a7b5fbf8089851150e3014d8fffc42dde76faee::m::S>"
                                  }
                                }
                              ]
                            }
                          ]
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 7, lines 151-186:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "asMovePackage": {
                      "module": {
                        "s": {
                          "module": {
                            "enum": null
                          }
                        },
                        "t": {
                          "module": {
                            "enum": {
                              "name": "T"
                            }
                          }
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
