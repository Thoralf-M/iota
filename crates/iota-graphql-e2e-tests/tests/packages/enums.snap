---
source: external-crates/move/crates/move-transactional-test-runner/src/framework.rs
---
processed 8 tasks

init:
A: object(0,0)

task 1, lines 7-18:
//# publish --upgradeable --sender A
created: object(1,0), object(1,1)
mutated: object(0,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 5768400,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, line 20:
//# create-checkpoint
Checkpoint created: 1

task 3, lines 22-67:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x10c3c7bd79faaeda6c85ce314060261a114292e5cf8c0bb626462dd82ea476b7",
                    "asMovePackage": {
                      "module": {
                        "enum": {
                          "name": "S",
                          "abilities": [
                            "COPY",
                            "DROP"
                          ],
                          "typeParameters": [],
                          "variants": [
                            {
                              "name": "V1",
                              "fields": [
                                {
                                  "name": "pos0",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            },
                            {
                              "name": "V2",
                              "fields": [
                                {
                                  "name": "x",
                                  "type": {
                                    "repr": "bool",
                                    "signature": {
                                      "ref": null,
                                      "body": "bool"
                                    }
                                  }
                                },
                                {
                                  "name": "y",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            }
                          ]
                        }
                      }
                    }
                  }
                },
                {
                  "outputState": {
                    "address": "0x61b03cabfaea4df64a748027dd8255715b996f453f572471b50537258a23eb16",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd35480b63fd276a2c1be5b7b6f24afe84ecd184ae8efde46979160968f760eeb",
                    "asMovePackage": null
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 4, lines 69-82:
//# upgrade --package P0 --upgrade-capability 1,0 --sender A
created: object(4,0)
mutated: object(0,0), object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 6642400,  storage_rebate: 2606800, non_refundable_storage_fee: 0

task 5, line 84:
//# create-checkpoint
Checkpoint created: 2

task 6, lines 86-149:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x61b03cabfaea4df64a748027dd8255715b996f453f572471b50537258a23eb16",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd35480b63fd276a2c1be5b7b6f24afe84ecd184ae8efde46979160968f760eeb",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xfecc5b14ea3a92e2730f4b738c78ec423ff078489c6d62bdd0bd1f3902e80d99",
                    "asMovePackage": {
                      "module": {
                        "s": {
                          "module": {
                            "package": {
                              "address": "0x10c3c7bd79faaeda6c85ce314060261a114292e5cf8c0bb626462dd82ea476b7"
                            }
                          },
                          "name": "S",
                          "abilities": [
                            "COPY",
                            "DROP"
                          ],
                          "typeParameters": [],
                          "variants": [
                            {
                              "name": "V1",
                              "fields": [
                                {
                                  "name": "pos0",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            },
                            {
                              "name": "V2",
                              "fields": [
                                {
                                  "name": "x",
                                  "type": {
                                    "repr": "bool",
                                    "signature": {
                                      "ref": null,
                                      "body": "bool"
                                    }
                                  }
                                },
                                {
                                  "name": "y",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                }
                              ]
                            }
                          ]
                        },
                        "t": {
                          "module": {
                            "package": {
                              "address": "0xfecc5b14ea3a92e2730f4b738c78ec423ff078489c6d62bdd0bd1f3902e80d99"
                            }
                          },
                          "name": "T",
                          "abilities": [],
                          "typeParameters": [
                            {
                              "constraints": [
                                "DROP"
                              ],
                              "isPhantom": false
                            }
                          ],
                          "variants": [
                            {
                              "name": "VV",
                              "fields": [
                                {
                                  "name": "y",
                                  "type": {
                                    "repr": "u64",
                                    "signature": {
                                      "ref": null,
                                      "body": "u64"
                                    }
                                  }
                                },
                                {
                                  "name": "s",
                                  "type": {
                                    "repr": "0x10c3c7bd79faaeda6c85ce314060261a114292e5cf8c0bb626462dd82ea476b7::m::S",
                                    "signature": {
                                      "ref": null,
                                      "body": {
                                        "datatype": {
                                          "package": "0x10c3c7bd79faaeda6c85ce314060261a114292e5cf8c0bb626462dd82ea476b7",
                                          "module": "m",
                                          "type": "S",
                                          "typeParameters": []
                                        }
                                      }
                                    }
                                  }
                                },
                                {
                                  "name": "u",
                                  "type": {
                                    "repr": "$0",
                                    "signature": {
                                      "ref": null,
                                      "body": {
                                        "typeParameter": 0
                                      }
                                    }
                                  }
                                }
                              ]
                            }
                          ]
                        },
                        "v": {
                          "name": "V",
                          "variants": [
                            {
                              "name": "V",
                              "fields": [
                                {
                                  "name": "t",
                                  "type": {
                                    "repr": "0x10c3c7bd79faaeda6c85ce314060261a114292e5cf8c0bb626462dd82ea476b7::m::T<0x10c3c7bd79faaeda6c85ce314060261a114292e5cf8c0bb626462dd82ea476b7::m::S>"
                                  }
                                }
                              ]
                            }
                          ]
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 7, lines 151-186:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "asMovePackage": {
                      "module": {
                        "s": {
                          "module": {
                            "enum": null
                          }
                        },
                        "t": {
                          "module": {
                            "enum": {
                              "name": "T"
                            }
                          }
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
